common/persistence/sql/sqlplugin/mysql/visibility.go

```diff
@@ -81,9 +81,9 @@ const (
 
 	templateGetClosedWorkflowExecutionsByStatus = templateClosedSelect + `AND status = ?` + templateConditionsClosedWorkflows
 
-	templateGetClosedWorkflowExecution = `SELECT workflow_id, run_id, start_time, execution_time, memo, encoding, close_time, workflow_type_name, status, history_length, task_queue 
+	templateGetWorkflowExecution = `SELECT workflow_id, run_id, start_time, execution_time, memo, encoding, close_time, workflow_type_name, status, history_length, task_queue 
 		 FROM executions_visibility
-		 WHERE namespace_id = ? AND status != 1
+		 WHERE namespace_id = ?
```

Comment:
this means the query could return result for open workflow, won't that be a problem for existing use cases?

Reply:
Reverted changes and add a new method specifically to get a single row from visibility.